### PR TITLE
User-to-JSON

### PR DESCRIPTION
**Pull request by sbocoun**
Wrote classes that handle outgoing data formatting for the Grade API.

Also, refactored package structure of the grade api subdirectory and added a default UserRating to the Movie initializer.

The outgoing data formatting classes consist of a CollectionBuilder, which takes the user's collections object, splits it into individual collections, and calls on individual media-JSON builder classes that convert each media type to JSON.

### CODE REVIEW

File: .idea/checkstyle-idea.xml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <component name="CheckStyle-IDEA" serialisationVersion="2">
     <checkstyleVersion>10.18.0</checkstyleVersion>
     <scanScope>JavaOnly</scanScope>
+    <scanBeforeCheckin>true</scanBeforeCheckin>
```
Comment (papertots) [inline .idea/checkstyle-idea.xml:6]: what does this do? I can't find documentation on this

Comment (sbocoun) [reply .idea/checkstyle-idea.xml:6]: It means Checkstyle automatically runs a code check before you "check in"/add your code to Git.

File: src/main/java/entity/User.java [hunk 2]
```diff
@@ -46,24 +40,29 @@ public void setMediaCollections(List<MediaCollection<? extends AbstractMedia>> m
...
+     * @param mediaType the type of media stored in the collection
      * @return a list of media collections
      */
-    @SuppressWarnings("unchecked")
-    public List<MediaCollection<Movie>> getMediaCollectionsMovies() {
-        final List<MediaCollection<Movie>> movieCollections = new ArrayList<>();
-        for (MediaCollection<? extends AbstractMedia> mediaCollection : mediaCollections) {
-            if (mediaCollection.getMediaType() == Movie.class) {
-                movieCollections.add((MediaCollection<Movie>) mediaCollection);
+    public <T> List<MediaCollection<T>> specifiedMediaCollections(Class<T> mediaType) {
```
Comment (papertots) [inline src/main/java/entity/User.java]: I would recommend renaming this to something like `getSpecifiedMediaCollection` so it's got a verb in the name

**Review by papertots — APPROVED** [review]
Reviewed before merge from main, everything looked good. Then resolved conflict from main. Should be good to merge now.

**Pull request merged**